MetaMask/metamask-extension
# feat: check for ledger error codes in the message and display a more …

## **Description**

This pr adds more user friendly error messages instead of just error codes when a user encounters ledger connection issues.

Resolves https://github.com/MetaMask/metamask-extension/issues/17606

## **Manual testing steps**

1. Go to Add hardware wallet
2. On the ledger device, unlock it.
3. On the `Connect a hardware wallet` page, Click on Ledger and then continue
4. Do not click connect when the webhid connection window pops up.
5. On the ledger device, lock the device
6. On the browser, click on the nano and then continue
7. See the new error


## **Screenshots/Recordings**

### **Before**

![ledger-error-2](https://github.com/MetaMask/metamask-extension/assets/96463427/67d44ae3-6967-415a-b2ae-bfb4e20911fd)


### **After**

![ledger-error](https://github.com/MetaMask/metamask-extension/assets/96463427/896edf82-57b4-4fe1-b04a-75afcfeed149)


## **Related issues**

Resolves https://github.com/MetaMask/metamask-extension/issues/17606

## **Pre-merge author checklist**


## **Pre-merge reviewer checklist**
